### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by yashodhicy**
In this milestone, we add classes that match the below UML class diagram.

catalog_of_my_things

🌺 Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.
🌺 All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
🌺 All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
🌺 add_item method (in all classes that have that method)

- [x] should take an instance of the Item class as an input.
- [x] should add the input item to the collection of items.
- [x] should add self as a property of the item object (by using the correct setter from the item object).

🌺can_be_archived?() in the Item class

- [x] should return true if published_date is older than 10 years.
- [x] Otherwise, it should return false.

🌺move_to_archive() in the Item class

- [x] should reuse can_be_archived?() method.
- [x] should change the archived property to true if the result of the can_be_archived?() method is true.
- [x] should do nothing if the result of the can_be_archived?() method is false.

🌺can_be_archived?() in the Book class

- [x] should override the method from the parent class.
- [x] should return true if parent's method returns true OR if cover_state equals to "bad".
- [x] otherwise, it should return false.

🌺can_be_archived?() in the MusicAlbum class

- [x] should override the method from the parent class.
- [x] should return true if parent's method returns true AND if on_spotify equals true.
- [x] otherwise, it should return false.

🌺can_be_archived?() in the Movie class

- [x] should override the method from the parent class.
- [x] should return true if parent's method returns true OR if silent equals true.
- [x] otherwise, it should return false.

🌺can_be_archived?() in the Game class

- [x] should override the method from the parent class.
- [x] should return true if parent's method returns true AND if last_played_at is older than 2 years.
- [x] otherwise, it should return false.

🌺Add unit tests for all implemented methods.

**User interface**

🌺Create a main.rb file that will serve as your console app entry point.

- [x] Present the user with a list of options to perform.
- [x] Let users choose an option.
- [x] If needed, ask for parameters for the option.
- [x] Have a way to quit the app.

🌺The following options should be available:

- List all books
- List all music albums
- List all movies
- List of games
- List all genres (e.g 'Comedy', 'Thriller')
- List all labels (e.g. 'Gift', 'New')
- List all authors (e.g. 'Stephen King')
- List all sources (e.g. 'From a friend', 'Online shop')
- Add a book
- Add a music album
- Add a movie
- Add a game

🌺All data should be preserved by saving collections in .json files.

**Database**
🌺Create a schema.sql file with tables that will be analogical to the structure of the classes in-app.

### CODE REVIEW

File: classes/MusicAlbum/music_album.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative '../item'
+
+class MusicAlbum < Item
+  attr_accessor :on_spotify
+
+  def initialize(published_date = nil, on_spotify: false, archived: false)
+    super(published_date, archived: archived)
+    @on_spotify = on_spotify
```
Comment (joel-kalema) [inline classes/MusicAlbum/music_album.rb:8]: - [ ] Good job implementing `MusicAlbum` class, however, currently we are not saving data for the `music_album` and `genres`.  Kindly  make sure to save data after adding the `music album`  and `genres` as it is [required](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#:~:text=All%20data%20should%20be%20preserved%20by%20saving%20collections%20in%20.json%20files.) to preserve data 👍 

Screenshot 2023-08-10 170806

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi TEAM
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 

- All tests are passing :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @yashodhicy, @Shaheryar0054 & @AliShahSafdari 

Great work on making the changes requested by a previous reviewer 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Data is preserved in JSON files
- ✅ All tests are passing
- ✅ Descriptive PR and README file
- ✅ Relationships are handled well

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  Data preserved properly. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**